chunky/src/java/se/llbit/math/structures/Position2ReferenceStructure.java

```diff
@@ -0,0 +1,11 @@
+package se.llbit.math.structures;
+
+public interface Position2ReferenceStructure<T> {
```

Comment:
I think a specialized Position2ColorStructure interface would lead more leeway to optimize the grass/foliage/water color in the future. With this generic interface, I don't how I could implement something that stores the colors in a flatten float[] (instead of in a float[][])

Reply:
It should work fine with this interface right? Can just copy into the falttened one. If we had varargs generics it might be better but it certainly works

Reply:
Yeah sorry, I was a bit confused about the fact that it was a generic interface.
In my head, because the interface was generic, the implementation had to be generic as well and I couldn't see how an implementation that is generic could take advantage of the flattening (because T could be something else than an array) but what I want is just a non generic implementation that implements Position2ReferenceStructure<float[]>.